packages/captp/src/trap.js

```diff
@@ -0,0 +1,87 @@
+// @ts-check
+// Lifted mostly from `@agoric/eventual-send/src/E.js`.
+
+import './types';
+
+/**
+ * Default implementation of Trap for near objects.
+ *
+ * @type {TrapImpl}
+ */
+export const nearTrapImpl = harden({
+  applyFunction(target, args) {
+    return target(...args);
+  },
+  applyMethod(target, prop, args) {
+    return target[prop](...args);
+  },
+  get(target, prop) {
+    return target[prop];
+  },
+});
+
+const readOnlyProxyHandler = {
```

Comment:
Wasn't there a review comment about missing traps to prevent modification? `defineProperty`. Also, did you mean `preventExtensions` rather than `isExtensible` below?

Reply:
I didn't have much intention; the E proxy maker implementation from which this was copied was written by more naive version of myself.  But your comments make sense so I've implemented them.

Reply:
Actually, trying to make these changes resulted in the following (I've reverted the changes but renamed this to `baseFreezableProxyHandler`):

```
TypeError: 'defineProperty' on proxy: trap returned falsish for property 'length'
    at freeze (<anonymous>)
```

```
TypeError: 'preventExtensions' on proxy: trap returned falsish
    at freeze (<anonymous>)
```

```
TypeError: 'preventExtensions' on proxy: trap returned truish but the proxy target is extensible
    at freeze (<anonymous>)
```

Reply:
Filed https://github.com/Agoric/agoric-sdk/issues/3494 . Resolving here.